pre-commit pylint hook doesn't use configuration file

The pre-commit-config file doesn't specify the configuration file(pylintrc) for pylint, so the pylint hook is running with its default settings.
This introduces a discrepancy between running pylint from pre-commit and github workflows.

The pre-commit-config for the pylint hook should be defined as:
```
- repo: local
  hooks:
  - id: pylint
    name: Check lint
    entry: pylint
    language: python
    files: ^src/package/|^tests/|setup.py
    args: [--rcfile=pylintrc]
```

Thanks @jovans-getpassport for opening the issue!

> The pre-commit-config file doesn't specify the configuration file(pylintrc) for pylint, so the pylint hook is running with its default settings. This introduces a discrepancy between running pylint from pre-commit and github workflows.

Are you saying that running `pylint` through pre-commit locally fails for you, whereas it succeeds when running in an Action; or the other way around?

Let me try something… When I remove this line:

https://github.com/jenstroeger/python-package-template/blob/1d7ab62da35cf2e211bd154832aa33293bde5cb5/src/package/something.py#L9

then I get the expected results in my local activated environment:
```
> pylint src
************* Module package.something
src/package/something.py:8:4: C0116: Missing function or method docstring (missing-function-docstring)

-------------------------------------------------------------------
Your code has been rated at 9.17/10 (previous run: 10.00/10, -0.83)
```
and
```
> pre-commit run pylint
Check lint...............................................................Failed
- hook id: pylint
- exit code: 16

************* Module package.something
src/package/something.py:8:4: C0116: Missing function or method docstring (missing-function-docstring)

-------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 10.00/10, -3.33)
```
Interesting that both runs produce a different score 🤔 When I then update the `pylintrc` file to filter the warning, both runs succeed:
```
> pylint src

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 9.17/10, +0.83)

> pre-commit run pylint
Check lint...............................................................Passed
```
Running `pylint` as an Action 

https://github.com/jenstroeger/python-package-template/blob/1d7ab62da35cf2e211bd154832aa33293bde5cb5/.github/workflows/pull-request.yaml#L48-L49

uses the repo, whereas (I think) running `pylint` as a [local hook](https://pre-commit.com/#repository-local-hooks) _may_ set a different working directory — although it’s worked above 🤔

So, looking at how `pylint` [searches for a configuration file](https://pylint.pycqa.org/en/latest/user_guide/run.html#command-line-options) then we can see the difference:
```
> pylint src

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

> cd src
src > pylint package
************* Module package.something
package/something.py:8:4: C0116: Missing function or method docstring (missing-function-docstring)
package/something.py:4:0: R0903: Too few public methods (1/2) (too-few-public-methods)

-------------------------------------------------------------------
Your code has been rated at 8.33/10 (previous run: 10.00/10, -1.67)

src > pylint --rcfile=../pylintrc package

-------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 8.33/10, +1.67)
```
whereas the hook still succeeds:
```
src > pre-commit run pylint
Check lint...............................................................Passed
```

> The pre-commit-config for the pylint hook should be defined as: […]

Adding that one line wouldn’t hurt, though in that case I’m tempted to check the other local hooks as well and add the configuration files explicitly too. @behnazh, what do you think?

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -66,7 +66,7 @@ repos:
     entry: pylint
     language: python
     files: ^src/package/|^tests/|setup.py
-    args: []
+    args: [--rcfile, pylintrc]
 
 # Type-check all Python code.
 - repo: local

